public/app/plugins/panel/geomap/layers/data/markersLayer.tsx

```diff
@@ -107,8 +125,17 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
             dot.setStyle(shape!.make(color, fillColor, radius));
             features.push(dot);
           };
-        }
 
+          // Post updates to the legend component
+          if (legend) {
+            console.log( 'UPDATE (marker layer)', colorDim);
+            legendProps.next({
+              color: colorDim,
+              size: sizeDim,
+            });
+          }
+          break; // Only the first frame for now!
```

Comment:
why only the first frame? Is this bug related, https://github.com/grafana/grafana/issues/48274 ? (credits to @yoyo421)